django/middleware/csrf.py

```diff
@@ -138,6 +138,10 @@ class CsrfViewMiddleware(MiddlewareMixin):
     """
     # The _accept and _reject methods currently only exist for the sake of the
     # requires_csrf_token decorator.
+
+    def __repr__(self):
+        return '<%s get_response=%s>' % (self.__class__.__qualname__, self.get_response.__qualname__)
```

Comment:
To avoid issues with class-based `get_response`, we should check if `__qualname__` is available e.g.
```suggestion
        return '<%s get_response=%s>' % (
            self.__class__.__qualname__,
            getattr(
                self.get_response,
                '__qualname__',
                self.get_response.__class__.__name__,
            ),
        )
```
I would also add test for this case
```python
        class GetResponse:
            def __call__(self):
                return HttpResponse()

        self.assertEqual(
            CsrfViewMiddleware(GetResponse()),
            '<CsrfViewMiddleware get_response=GetResponse>',
        )
```

Reply:
We should also move `__repr__()` to `MiddlewareMixin` which will solve this ticket for other middlewares: `RedirectFallbackMiddleware`, `MessageMiddleware`, `RemoteUserMiddleware` etc.

Reply:
Is there any further tests needed for `RedirectFallbackMiddleware`, `MessageMiddleware` and `RemoteUserMiddleware`?

Reply:
We can add tests with `self.subTest()` to the `tests/deprecation/test_middleware_mixin.py`.